core/engine/src/builtins/temporal/plain_year_month/mod.rs

```diff
@@ -338,3 +465,84 @@ pub(crate) fn create_temporal_year_month(
     // 9. Return object.
     Ok(obj.into())
 }
+
+// 9.5.9 AddDurationToOrSubtractDurationFromPlainYearMonth ( operation, yearMonth, temporalDurationLike, options )
+fn add_or_subtract_duration(
+    is_addition: bool,
+    this: &JsValue,
+    duration_like: &JsValue,
+    options: &JsObject,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    let duration: Duration = if duration_like.is_object() {
+        to_temporal_duration(duration_like, context)?
+    } else if duration_like.is_string() {
```

Comment:
Same as mentioned elsewhere. `if let Some(s) = item.as_string()` would ideally be used here. Doing so will remove some panics.